### PR TITLE
Fix S5944 FP: When namespace or type is same as function name

### LINKED ISSUE #4347: Fix S5944 FP: When namespace or type is same as function name
### Description

Rule raises issue when namespace is the same as function name.

Rule raises issue when return type is the same as function name.

Rule raises issue when declared type is same as function name

### Repro steps

```
Public Function Data() As Data.Container 'S5944	Do not make use of the implicit return value.
    Dim D As Data.Container    'S5944	Do not make use of the implicit return value.
End Function    

Public Function Data() As Data  'S5944	Do not make use of the implicit return value.
    Dim D As Data  'S5944	Do not make use of the implicit return value.
End Function
```

* SonarLint 4.33.1
* Visual Studio version 2019

### PR DESCRIPTION
**Pull request by pavel-mikula-sonarsource**
Fixes #4347

### CODE REVIEW

**Comment by pavel-mikula-sonarsource** [comment]
The PR is missing [coverage info on SC](https://sonarcloud.io/summary/new_code?id=sonaranalyzer-dotnet&pullRequest=4497) for some reason

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/UseReturnStatement.vb [hunk 1]
```diff
@@ -124,6 +139,20 @@ Public Class DoesNotApplyOn
...
+        Dim Ret As OtherType                    ' Noncompliant FP
+    End Function
+
+    Public Function NamespaceName() As NamespaceName.Something  'Noncompliant FP
+        Dim Ret As NamespaceName.Something                      'Noncompliant FP
+    End Function
+
+    Public Sub SomeEvent() Handles fSource.SomeEvent
+        Dim S As Source
+        AddHandler S.SomeEvent, AddressOf SomeEvent
```
Comment (andrei-epure-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/TestCases/UseReturnStatement.vb:158]: I don't get it - is this part of the repro?

**Review by andrei-epure-sonarsource — APPROVED** [review]
LGTM. I left a comment for me to better understand. Not sure it requires a change or not.

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/UseReturnStatement.vb (hunk 1 above)
Comment (pavel-mikula-sonarsource) [reply analyzers/tests/SonarAnalyzer.UnitTest/TestCases/UseReturnStatement.vb:158]: No, this is just another unrelated syntax that came to my mind while working on it and I wanted to be sure we don't have another FP there.

**Pull request merged**